Latest release/Archive pages now accept any platform, plus fixes

Joe-Brady opened the pull request:
Related to #35 

CURRENTLY ON STAGING: http://staging.adoptopenjdk.net/releases.html

Completed the second step towards full abstraction in the JS to accept any defined platform.

The 'Latest' section of releases.js will now accept all platform that are defined in the 'platforms' array at the top of global.js.

**Note:** new platforms are just added as blocks to the Latest page. These blocks appear in a long row. This long row flows onto new lines when it reaches the defined maximum width on the page. I will add a low-priority issue to possibly change this layout in the future, because while it works, 4/5/6+ platforms could perhaps be displayed more efficiently.

gdams commented:
@Joe-Brady can you add the s390x and ppc64le icons ?

karianna commented on dist/js/app.js:
```diff
@@ -309,10 +326,10 @@ function populateNightly() {
             // (this filters out all non-binary attachments, e.g. SHA checksums - these contain the platform name, but are not binaries)
+            var thisFileExtension = getFileExt(thisPlatform); // get the file extension associated with this platform
             if(a.indexOf((thisFileExtension.toUpperCase())) >= 0) {
 
```
I know you really, really want rename `a` :-)

karianna commented on dist/js/app.js:
```diff
@@ -484,54 +501,49 @@ function populateLatest() {
-
+      // for each asset attached to this release, check if it's a valid binary, then add a download block for it...
       assetCounter2 = 0;
-      assetArray.forEach(function() {     // iterate through the binaries attached to this release
```
Usually when you have loop counters like `assetCounter` and a `assetCounter2` it's a scoping or naming code smell....

Joe-Brady replied:
Opened issue #65 for this

karianna commented on dist/js/app.js:
```diff
@@ -484,54 +501,49 @@ function populateLatest() {
-          macDlButton.href = macLink;
-          macPlatformBlock.className = macPlatformBlock.className.replace( /(?:^|\s)hide(?!\S)/g , '' );
+        var a = nameOfFile.toUpperCase(); // make the name of the asset uppercase
+        var thisPlatform = getSearchableName(a); // get the searchableName, e.g. MAC or X64_LINUX.
```
You know what I'm going to say here :-)

Joe-Brady commented:
@gdams @karianna - Thanks for the reviews, I'll fix those bits as part of some other incoming changes to this PR! (Please don't merge this PR yet)

Joe-Brady commented:
@gdams @karianna 
Ok, so I've done a lot of changes since the last time you checked:

For most of the changes, check out the names of each commit. Most notable is the (now fully populated) Archive page, with a brand-new, shiny platform filter drop-down.

Just to also note that the var 'a' is now named properly, and George's request for the logos is also complete.
